exchanges/deribit/deribit_test.go

```diff
@@ -0,0 +1,3034 @@
+	require.NoError(t, err)
+}
+
+func TestFetchOrderbook(t *testing.T) {
+	t.Parallel()
+	_, err := d.FetchOrderbook(context.Background(), futuresTradablePair, asset.Futures)
+	require.NoError(t, err)
+	_, err = d.FetchOrderbook(context.Background(), spotTradablePair, asset.Spot)
+	require.NoError(t, err)
+	sleepUntilTradablePairsUpdated()
+	_, err = d.FetchOrderbook(context.Background(), futureComboTradablePair, asset.FutureCombo)
+	require.NoError(t, err)
+	_, err = d.FetchOrderbook(context.Background(), optionComboTradablePair, asset.OptionCombo)
+	require.NoError(t, err)
+	_, err = d.FetchOrderbook(context.Background(), optionsTradablePair, asset.Options)
+	assert.NoError(t, err)
+}
+
+func TestUpdateAccountInfo(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d)
+	_, err := d.UpdateAccountInfo(context.Background(), asset.Futures)
+	assert.NoError(t, err)
+}
+
+func TestFetchAccountInfo(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d)
+	_, err := d.FetchAccountInfo(context.Background(), asset.Spot)
+	assert.NoError(t, err)
+	_, err = d.FetchAccountInfo(context.Background(), asset.Futures)
+	assert.NoError(t, err)
+	_, err = d.FetchAccountInfo(context.Background(), asset.Options)
+	assert.NoError(t, err)
+	_, err = d.FetchAccountInfo(context.Background(), asset.OptionCombo)
+	assert.NoError(t, err)
+	_, err = d.FetchAccountInfo(context.Background(), asset.FutureCombo)
+	assert.NoError(t, err)
+}
+
+func TestGetFundingHistory(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d)
+	_, err := d.GetAccountFundingHistory(context.Background())
+	assert.NoError(t, err)
+}
+
+func TestGetWithdrawalsHistory(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d)
+	_, err := d.GetWithdrawalsHistory(context.Background(), currency.BTC, asset.Empty)
+	assert.NoError(t, err)
+}
+
+func TestGetRecentTrades(t *testing.T) {
+	t.Parallel()
+	_, err := d.GetRecentTrades(context.Background(), futuresTradablePair, asset.Futures)
+	assert.NoError(t, err)
+	_, err = d.GetRecentTrades(context.Background(), spotTradablePair, asset.Spot)
+	assert.NoError(t, err)
+}
+
+func TestWSRetrievePublicPortfolioMargins(t *testing.T) {
+	t.Parallel()
+	info, err := d.GetInstrument(context.Background(), btcPerpInstrument)
+	if err != nil {
+		t.Skip(err)
+	}
+	_, err = d.WSRetrievePublicPortfolioMargins(currencyBTC, map[string]float64{btcPerpInstrument: info.ContractSize * 2})
+	assert.NoError(t, err)
+}
+
+func TestCancelAllOrders(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d, canManipulateRealOrders)
+	var orderCancellation = &order.Cancel{
+		OrderID:       "1",
+		WalletAddress: core.BitcoinDonationAddress,
+		AccountID:     "1",
+		Pair:          futuresTradablePair,
+		AssetType:     asset.Futures,
+	}
+	_, err := d.CancelAllOrders(context.Background(), orderCancellation)
```

Comment:
```
    deribit_test.go:2382: 
        	Error Trace:	/Users/scottg/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/deribit/deribit_test.go:2382
        	Error:      	Should be false
        	Test:       	TestCancelAllOrders
        	Messages:   	json: cannot unmarshal array into Go value of type deribit.CancelResponse
```

Use the testnet to test this one

Reply:
They have updated this endpoint and it is returning an integer .. That's the reason. I have updated it anyways

Reply:
Do you mean that the testnet is outdated in its response?
```
Deribit raw response: {"jsonrpc":"2.0","result":[],"usIn":1714350206341039,"usOut":1714350206341877,"usDiff":838,"testnet":true}
    deribit_test.go:2393: 
        	Error Trace:	/Users/s/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/deribit/deribit_test.go:2393
        	Error:      	Should be false
        	Test:       	TestCancelAllOrders
        	Messages:   	json: cannot unmarshal array into Go value of type int64
Deribit attempt 1 request path: https://test.deribit.com/api/v2/private/cancel_all_by_instrument?detailed=true&include_combos=true&instrument_name=BTC-STRD-28JUN24-64000&type=all
Deribit request header [Content-Type]: [application/json]
Deribit request header [Authorization]: [deri-hmac-sha256 id=jo4AImtw,ts=1714350206207,sig=8171af5c98375277e9939bc143e9cd5198498a191e751d33e6573e493f611100,nonce=1714350199141528004]
Deribit request type: GET
HTTP status: 200 OK, Code: 200
Deribit raw response: {"jsonrpc":"2.0","result":[],"usIn":1714350206700921,"usOut":1714350206701471,"usDiff":550,"testnet":true}
    deribit_test.go:2397: 
        	Error Trace:	/Users/s/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/deribit/deribit_test.go:2397
        	Error:      	Should be false
        	Test:       	TestCancelAllOrders
        	Messages:   	json: cannot unmarshal array into Go value of type int64
--- FAIL: TestCancelAllOrders (7.14s)
```

Reply:
I could not test it on the main net as there is no fund in my account, however, the API documentation is updated and has an integer value as a response instead of an object or list of objects.

Reply:
<img width="868" alt="Screenshot 2024-04-29 at 4 06 52 PM" src="https://github.com/thrasher-corp/gocryptotrader/assets/39623015/ad479df7-9236-45bd-8b3a-7a4c9a4994de">